### PR TITLE
Create empty completion list if java ls completion is null

### LINKED ISSUE #347: Many Exceptions spit out when typing in a Java file
I get the following Exception when typing in a Java file. It makes it really hard to code, since in VS Code it makes the output pop up on every keystroke.

```
java.util.concurrent.CompletionException: java.lang.NullPointerException: Cannot invoke "org.eclipse.lsp4j.CompletionList.setIsIncomplete(boolean)" because "list" is null
	at java.base/java.util.concurrent.CompletableFuture.encodeThrowable(Unknown Source)
	at java.base/java.util.concurrent.CompletableFuture.completeThrowable(Unknown Source)
	at java.base/java.util.concurrent.CompletableFuture$UniApply.tryFire(Unknown Source)
	at java.base/java.util.concurrent.CompletableFuture.postComplete(Unknown Source)
	at java.base/java.util.concurrent.CompletableFuture.complete(Unknown Source)
	at org.eclipse.lsp4j.jsonrpc.RemoteEndpoint.handleResponse(RemoteEndpoint.java:212)
	at org.eclipse.lsp4j.jsonrpc.RemoteEndpoint.consume(RemoteEndpoint.java:193)
	at org.eclipse.lsp4mp.ls.commons.ParentProcessWatcher.lambda$apply$0(ParentProcessWatcher.java:148)
	at org.eclipse.lsp4j.jsonrpc.json.StreamMessageProducer.handleMessage(StreamMessageProducer.java:194)
	at org.eclipse.lsp4j.jsonrpc.json.StreamMessageProducer.listen(StreamMessageProducer.java:94)
	at org.eclipse.lsp4j.jsonrpc.json.ConcurrentMessageProcessor.run(ConcurrentMessageProcessor.java:113)
	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Unknown Source)
	at java.base/java.util.concurrent.FutureTask.run(Unknown Source)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)
	at java.base/java.lang.Thread.run(Unknown Source)
```

### PR DESCRIPTION
**Pull request by datho7561**
Fixes #347

Signed-off-by: David Thompson <davthomp@redhat.com>

### CODE REVIEW

File: microprofile.ls/org.eclipse.lsp4mp.ls/src/main/java/org/eclipse/lsp4mp/ls/java/JavaFileTextDocumentService.java [hunk 1]
```diff
@@ -160,6 +160,11 @@ public CompletableFuture<Either<List<CompletionItem>, CompletionList>> completio
 				cancelChecker.checkCanceled();
 
 				CompletionList list = completionResult.getCompletionList();
+				if (list == null) {
+					list = new CompletionList();
+				}
+				CompletionList finalizedList = list;
```
Comment (fbricon) [inline microprofile.ls/org.eclipse.lsp4mp.ls/src/main/java/org/eclipse/lsp4mp/ls/java/JavaFileTextDocumentService.java:166]: this seems unnecessary

File: microprofile.ls/org.eclipse.lsp4mp.ls/src/main/java/org/eclipse/lsp4mp/ls/java/JavaFileTextDocumentService.java [hunk 2]
```diff
@@ -172,13 +177,13 @@ public CompletableFuture<Either<List<CompletionItem>, CompletionList>> completio
 							return true;
 						}, projectInfo) //
 						.forEach(item -> {
-							list.getItems().add(item);
+							finalizedList.getItems().add(item);
```
Comment (fbricon) [inline microprofile.ls/org.eclipse.lsp4mp.ls/src/main/java/org/eclipse/lsp4mp/ls/java/JavaFileTextDocumentService.java:180]: does new CompletionList() guarantee getItems() is not null?

**Review by fbricon — CHANGES_REQUESTED** [review]

File: microprofile.ls/org.eclipse.lsp4mp.ls/src/main/java/org/eclipse/lsp4mp/ls/java/JavaFileTextDocumentService.java (hunk 2 above)
Comment (datho7561) [reply microprofile.ls/org.eclipse.lsp4mp.ls/src/main/java/org/eclipse/lsp4mp/ls/java/JavaFileTextDocumentService.java:180]: Yes

File: microprofile.ls/org.eclipse.lsp4mp.ls/src/main/java/org/eclipse/lsp4mp/ls/java/JavaFileTextDocumentService.java (hunk 1 above)
Comment (datho7561) [reply microprofile.ls/org.eclipse.lsp4mp.ls/src/main/java/org/eclipse/lsp4mp/ls/java/JavaFileTextDocumentService.java:166]: Yeah, the functional programming makes the logic messy, I'll fix it

File: microprofile.ls/org.eclipse.lsp4mp.ls/pom.xml [hunk 3]
```diff
@@ -25,8 +25,8 @@
 
 	<properties>
 		<project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
-		<maven.compiler.target>1.8</maven.compiler.target>
-		<maven.compiler.source>1.8</maven.compiler.source>
+		<maven.compiler.target>11</maven.compiler.target>
```
Comment (datho7561) [inline microprofile.ls/org.eclipse.lsp4mp.ls/pom.xml:28]: oops

**Review by fbricon — APPROVED** [review]

**Pull request merged**